frontend/src/views/GeneratePosterPage/index.vue

```diff
@@ -281,10 +281,13 @@ export default {
       return validators
     },
     userCanteens() {
-      const canteens = this.$store.state.userCanteenPreviews
-      return canteens.sort((a, b) => {
-        return normaliseText(a.name) > normaliseText(b.name) ? 1 : 0
-      })
+      if (this.isAuthenticated) {
+        const canteens = this.$store.state.userCanteenPreviews
+        return canteens.sort((a, b) => {
+          return normaliseText(a.name) > normaliseText(b.name) ? 1 : 0
+        })
+      }
+      return []
```

Comment:
J'ai une préférence pour un early-return (par ex `if (!this.isAuthenticated) return []`). Mais étant une question de style je te laisse décider et j'approuve quand même.

Reply:
je suis d'acc